# Question 1

Built Docker image using

docker build -t zoomcamp-model:hw10 .

and then ran it using

docker run -it --rm -p 9696:9696 zoomcamp-model:hw10

Now we run python q6_test.py:

In [1]:
!python q6_test.py

{'get_credit': True, 'get_credit_probability': 0.726936946355423}


<b> Answer 1: 0.7269 </b>

# Question 2
What's the version of kind that you have?

Use kind --version to find out.

In [3]:
!kind --version

kind version 0.20.0


<b> Answer 2: 0.20.0 </b>

# Question 3
Now let's test if everything works. Use kubectl to get the list of running services.

What's CLUSTER-IP of the service that is already running there?

In [4]:
!kubectl get service

NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   3d4h


<b> Answer 3: 10.96.0.1 </b>

# Question 4
To be able to use the docker image we previously created (zoomcamp-model:hw10), we need to register it with kind.


What's the command we need to run for that?

kind create cluster <br>
kind build node-image <br>
kind load docker-image <br>
kubectl apply

<b> Answer 4: kind load docker-image </b>

# Question 5
Now let's create a deployment config (e.g. deployment.yaml):

Replace Image, Memory, CPU, Port with the correct values.

What is the value for Port?

Apply this deployment using the appropriate command and get a list of running Pods. You can see one running Pod.

<b> Answer 5: 9696 </b> (also see filled in deployment.yaml in the folder)

# Question 6
Let's create a service for this deployment (service.yaml):

What do we need to write instead of ???

<b> Answer 6: credit </b> (also see filled in service.yaml in the folder)